sites/example-project/src/components/ui/CodeBlock.svelte

```diff
@@ -0,0 +1,121 @@
+<script>
+  export let source;
+  export let copyToClipboard = true;
+  import Copy from "./Deployment/CopyIcon.svelte";
+  import Success from "./Deployment/CopySuccessIcon.svelte";
+  let copied = false;
+
+  const toggleCopied = function () {
+    copied = false;
+  };
+
+  export let copy = async (text) => {
+    try {
+      if (!copied) {
+        await navigator.clipboard.writeText(text);
+        copied = true;
+        setTimeout(toggleCopied, 1500);
+      }
+    } catch (e) {}
+  };
+</script>
+
+<pre><code>{#if source}{source}{:else}<slot />{/if}</code>
```

Comment:
Why do we want to put a `<slot/>`  if `source` is `undefined` ? 

if you don't need the `slot` you can start the component with

```svelte

{#if source !== undefined} 

```

or 

```svelte

{#if source }

```

and you will not have the need to validate `source` is not `undefined` in the inner-components

Reply:
The idea here would be to allow an alternative use of the codeblock like this:
```svelte
<CodeBlock language=html>
   <div>Hello world</div>
</CodeBlock>
```

as an alternative to the markdown syntax